core/src/main/java/hudson/security/Permission.java

```diff
@@ -274,22 +233,52 @@ public void setEnabled(boolean enable) {
     public boolean getEnabled() {
         return enabled;
     }
-    
+
+    @Override public int compareTo(@Nonnull Permission o) {
+        return getId().compareTo(o.getId());
+    }
+
+    /**
+     * Comparator that orders {@link Permission} objects based on their ID.
+     */
+    @Deprecated
+    public static final Comparator<Permission> ID_COMPARATOR = Comparator.comparing(Permission::getId);
```

Comment:
or just the natural ordering

Reply:
Oh right, I forgot that I added Comparable to Permission. That was mainly due to a JEP-200 issue with serializing lambdas.